### PR TITLE
add env var `LOGWATCH_SENDER`

### PR DESCRIPTION
**Pull request by craue**
# Description

Follow-up to #2360, but using `/etc/logwatch/conf/logwatch.conf` as config file.

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

This *might* be considered to be a breaking change if someone relied on the fact that `REPORT_SENDER` wouldn't influence the sender address for logwatch reports. But when I started using this project, I was actually expecting it would.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by craue** [comment]
I've fixed the issue's description because I just noticed that `POSTMASTER_ADDRESS` wouldn't bubble to `LOGWATCH_SENDER` or `PFLOGSUMM_SENDER`. The documenation for `PFLOGSUMM_SENDER` was already wrong and I just copied it. Shall I fix the comments for both sender vars?

File: docs/content/config/environment.md [hunk 1]
```diff
@@ -286,6 +286,13 @@ Recipient address for logwatch reports if they are enabled.
 - **not set** => Use REPORT_RECIPIENT or POSTMASTER_ADDRESS
 - => Specify the recipient address(es)
 
+##### LOGWATCH_SENDER
+
+From address for logwatch reports if they are enabled.
```
Comment (georglauterbach) [inline docs/content/config/environment.md:291]: I'd rephrase this:

This variable sets the sender address (`FROM`) for logwatch reports if logwatch reports are enabled.

**Comment by georglauterbach** [comment]
> I've fixed the issue's description because I just noticed that `POSTMASTER_ADDRESS` wouldn't bubble to `LOGWATCH_SENDER` or `PFLOGSUMM_SENDER`. The documenation for `PFLOGSUMM_SENDER` was already wrong and I just copied it.

What do you mean by "bubble"?

> Shall I fix the comments for both sender vars?

Sure :)

File: docs/content/config/environment.md (hunk 1 above)
Comment (craue) [reply docs/content/config/environment.md:291]: I also just copied these lines from `PFLOGSUMM_SENDER`. 😏

How about a bit shorter form?
**Sender address (`FROM`) for logwatch reports if logwatch reports are enabled.**

**Comment by craue** [comment]
> > I've fixed the issue's description because I just noticed that `POSTMASTER_ADDRESS` wouldn't bubble to `LOGWATCH_SENDER` or `PFLOGSUMM_SENDER`. The documenation for `PFLOGSUMM_SENDER` was already wrong and I just copied it.
> 
> What do you mean by "bubble"?

I mean that the value of a variable will also set another variable if it's not explicitly set otherwise.

> Sure :)

Will do.

**Comment by casperklein** [comment]
Don't forget to adjust `start-mailserver.sh` as well. As of now, the new env can't be used.

File: docs/content/config/environment.md (hunk 1 above)
Comment (georglauterbach) [reply docs/content/config/environment.md:291]: Fine with me :)

File: docs/content/config/environment.md [hunk 2]
```diff
@@ -286,6 +286,13 @@ Recipient address for logwatch reports if they are enabled.
...
+
+- **not set** => Use REPORT_SENDER or POSTMASTER_ADDRESS
```
Comment (georglauterbach) [inline docs/content/config/environment.md:293]: Maybe we should make sure which one it actually is :)

**Comment by georglauterbach** [comment]
> Don't forget to adjust `start-mailserver.sh` as well. As of now, the new env can't be used.

Isn't it enough that they're set up in `_setup_default_vars` in `setup-stack.sh`?

File: docs/content/config/environment.md (hunk 2 above)
Comment (craue) [reply docs/content/config/environment.md]: Only `REPORT_SENDER` is used, at least from what I can tell from the code. I've fixed the docs.

File: docs/content/config/environment.md (hunk 1 above)
Comment (craue) [reply docs/content/config/environment.md]: Done.

**Comment by casperklein** [comment]
I think you are right. I find it always confusing, when default vars are defined in two places 😕

**Comment by georglauterbach** [comment]
> I think you are right. I find it always confusing, when default vars are defined in two places confused

Very true. I guess the variables in `_setup_default_vars` need some setup beforehand (like hostname and stuff). That#s why they're located in that specific place.

**Comment by craue** [comment]
Not sure about the failing test. Is it related to my changes?

**Review by polarathene — APPROVED** [review]

**Review by georglauterbach — APPROVED** [review]

**Review by wernerfred — APPROVED** [review]

**Comment by georglauterbach** [comment]
> Not sure about the failing test. Is it related to my changes?
> 
> 
> 
> Edit: Interesting... now the failure is gone. 😮

Sometimes, although not too often, a test is flakey and does not behave well. Restarting the jobs solves these problems.
